contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+    mapping(address => uint256) public override delegatorWithdrawRequestedCTokenCount;
+
+    mapping(uint256 => DelegatorWithdrawInfo) public override delegatorWithdrawRequests;
+    mapping(address => uint256[]) public override requestIdsByDelegatorAddress;
+    mapping(address => uint256) private liquidationIndexByOperator;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    //TODO sanjay set variables with right value
+    function initialize(address _admin, address _staderConfig) external initializer {
+        UtilLib.checkNonZeroAddress(_admin);
+        UtilLib.checkNonZeroAddress(_staderConfig);
+        __AccessControl_init_unchained();
+        __Pausable_init();
+        staderConfig = IStaderConfig(_staderConfig);
+        utilizeIndex = DECIMAL;
+        utilizationRatePerBlock = 38051750380; // 10%
+        protocolFeeFactor = 0;
+        nextRequestId = 1;
+        nextRequestIdToFinalize = 1;
+        finalizationBatchLimit = 50;
+        undelegationPeriodInBlocks = 50400; //7 days
+        accrualBlockNumber = block.number;
+        minBlockDelayToFinalizeRequest = 14400; //2 days
+        maxNonRedeemedDelegatorRequestCount = 1000;
+        maxETHWorthOfSDPerValidator = 1 ether;
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+        emit UpdatedStaderConfig(_staderConfig);
+    }
+
+    /**
+     * @notice Sender delegate SD and cToken balance increases for sender
+     * @dev Accrues fee whether or not the operation succeeds, unless reverted
+     * @param sdAmount The amount of SD token to delegate
+     */
+    function delegate(uint256 sdAmount) external override whenNotPaused {
+        accrueFee();
+        ISDIncentiveController(staderConfig.getSDIncentiveController()).onDelegate(msg.sender);
+        _delegate(sdAmount);
+    }
+
+    /**
+     * @notice auxiliary method to put a withdrawal request, takes in cToken amount in input
+     * @param _cTokenAmount amount of cToken
+     * @return _requestId generated request ID for withdrawal
+     */
+    function requestWithdraw(uint256 _cTokenAmount) external override whenNotPaused returns (uint256 _requestId) {
+        if (_cTokenAmount > delegatorCTokenBalance[msg.sender] - delegatorWithdrawRequestedCTokenCount[msg.sender]) {
+            revert InvalidAmountOfWithdraw();
+        }
+        delegatorWithdrawRequestedCTokenCount[msg.sender] += _cTokenAmount;
+        accrueFee();
+        uint256 exchangeRate = _exchangeRateStoredInternal();
+        uint256 sdRequested = (exchangeRate * _cTokenAmount) / DECIMAL;
+        _requestId = _requestWithdraw(sdRequested, _cTokenAmount);
+    }
+
+    /**
+     * @notice auxiliary method to put a withdrawal request, takes SD amount in input
+     * @param _sdAmount amount of SD to withdraw
+     * @return _requestId generated request ID for withdrawal
+     */
+    function requestWithdrawWithSDAmount(uint256 _sdAmount)
```

Comment:
Why offer this function? Can't guarantee this withdrawal anyway if slashing or ER fluctuation occurs

Reply:
yes we can't guarantee but on dapp we will be showing the delegator SD balance, it will be convenient for them to withdraw by providing the SD token amount in the input